assets/src/js/app.js

```diff
@@ -1,36 +1,53 @@
 import stickyEl from './lib/stickyEl';
-
 const componentName = 'anchor-navigation';
 const baseSelector = '.' + componentName;
 const triggerSelector = baseSelector + '__trigger';
-const navEl = document.querySelector('section' + baseSelector);
-window.AnchorNavigation = {};
-window.AnchorNavigation.settings = null;
-window.AnchorNavigation.init = once(() => {
-  buildNavigation();
-});
-
-if (navEl && !navEl.dataset.attached) {
-  window.AnchorNavigation.settings = drupalSettings.anchorNavigation;
-  window.AnchorNavigation.init();
-  navEl.dataset.attached = true;
+let navEls = [];
+
+initAnchorNavigation();
+
+window.addEventListener('initAnchorNavigation', initAnchorNavigation);
+
+function initAnchorNavigation() {
+  debugger;
+  window.AnchorNavigation = [];
+  navEls = [...document.querySelectorAll('section' + baseSelector)];
```

Comment:
```suggestion
  navEls.push(...Array.from(document.querySelectorAll('section' + baseSelector));
```
You can spread the query selector result to the push method
Array.from is used to ensure it doesn't fail if the query is empty since the returned value is not an array but a nodelist